### ISSUE TITLE
"Cannot compile modules unless '--module' flag is provided" error even when "module" property is specified in tsconfig.json

### ISSUE THREAD

**Issue by nfriend** [Microsoft/vscode#2719]
I'm getting a `Cannot compile modules unless '--module' flag is provided` error in Visual Studio Code v. 0.10.6:

<img src="http://i.stack.imgur.com/qi0M4.png">

Here's my `tsconfig.json`, which is at the root of the directory open in Visual Studio Code.  Note that I have provided a value for the `module` property:

    {
        "compilerOptions": {
            "target": "es5",
            "module": "system",
            "noImplicitReturns": true,
            "noImplicitAny": true,
            "noFallthroughCasesInSwitch": true,
            "preserveConstEnums": true,
            "sourceMap": true,
            "outDir": "./src/scripts/compiled",
            "rootDir": "./src/scripts"
        },
        "exclude": [
            "node_modules",
            "jspm_packages"
        ]
    }

I've tried other values for the property, such as `amd` and `commonjs` without any luck.  I've also restarted the IDE several times.

The code compiles with no errors when I compile from the command line.  (I'm using [gulp-typescript](https://www.npmjs.com/package/gulp-typescript) for TypeScript compilation.

I have TypeScript 1.8 beta installed globally via `npm`, although this problem was also present when TypeScript 1.7 was installed.

**Comment by moostad** [comment]
Just remove ```noImplicitReturns": true,``` from your ```tsconfig.json```

**Comment by nfriend** [comment]
@moostad Just tried, no luck.

**Comment by moostad** [comment]
@nfriend You need to reload project, did you?

**Comment by nfriend** [comment]
@moostad Yep, I restarted Code - same result.

**Comment by moostad** [comment]
@nfriend I just removed both 
```
        "noImplicitReturns": true,
        "noFallthroughCasesInSwitch": true,
``` and problem solved however this is a problem with TsServer and not VScode.

**Comment by nfriend** [comment]
@moostad You're right - I removed all compiler options introduced in 1.8, and now it's working as expected.  This also solved a few other errors I was getting in the IDE.  I'm guessing this will be fixed automatically when 1.8 comes out of beta.

Thanks a ton!

**Comment by egamma** [comment]
You can also configure VS Code to use another version of typescript https://code.visualstudio.com/docs/languages/typescript#_using-newer-typescript-versions
